src/schema.ts

```diff
@@ -360,17 +364,17 @@ UserSchema.statics.getUser = async function ({ username, phone }) {
   return user
 }
 
-UserSchema.statics.getUserByAddress = async function ({ address }) {
-  return this.findOne({ "onchain.address": address })
-}
-
-// FIXME: Merge findByUsername and getUser
-UserSchema.statics.findByUsername = async function ({ username }) {
-  if (typeof username !== "string" || !username.match(regexUsername)) {
-    return null
+UserSchema.statics.findOrCreate = async function (userInfo) {
```

Comment:
are you sure this not change the current behavior of the app? the tests does not include much about this

Reply:
We changed all the places that were using this

Reply:
Found a few places where this was in issue. Solved in most recent commit.

Reply:
Hmm I consider having a `findOrCreate` method somewhat of an anti-pattern. Typically the code that would call this should have enough context to know which use-case is being handled. Also as far as I can tell this method is only called from `test` code and having a function in the production code that is only for testing purposes should also be avoided.

Can we move this to be a test helper?